minio/signer.py

```diff
@@ -198,7 +198,7 @@ def sign_v4(method, url, region, headers=None, access_key=None,
 
     date = datetime.utcnow()
     headers['X-Amz-Date'] = date.strftime("%Y%m%dT%H%M%SZ")
-    headers['X-Amz-Content-Sha256'] = content_sha256
+    headers['X-Amz-Content-Sha256'] = content_sha256.decode('ascii')
```

Comment:
content_sha256 can be "UNSIGNED_PAYLOAD" ".decode will fail in that scenario.  Do it like this 

```
if content_sha256 == _UNSIGNED_PAYLOAD: 
     headers['X-Amz-Content-Sha256'] = content_sha256
else:
     headers['X-Amz-Content-Sha256'] = content_sha256.decode('ascii')
```